FWCore/ParameterSet/src/FileInPath.cc

```diff
@@ -372,16 +370,17 @@ namespace edm {
     for (auto const& element : pathElements) {
       // Set the boost::fs path to the current element of
       // CMSSW_SEARCH_PATH:
-      bf::path pathPrefix(element);
+      std::filesystem::path pathPrefix(element);
 
       // Does the a file exist? locateFile throws is it finds
       // something goofy.
       if (locateFile(pathPrefix, relativePath_)) {
         // Convert relative path to canonical form, and save it.
-        relativePath_ = bf::path(relativePath_).normalize().string();
+        relativePath_ = std::filesystem::weakly_canonical(std::filesystem::path(relativePath_)).string();
```

Comment:
@davidlange6 , problem is here. For `relativePath_=./Geometry/CMSCommonData/data/materials.xml`   (note the `./` at the start) the `std::filesystem::weakly_canonical` returns `$PWD/Geometry/CMSCommonData/data/materials.xml` instead of `Geometry/CMSCommonData/data/materials.xml`